src/core/thread/mesh_forwarder.cpp

```diff
@@ -1014,6 +1033,18 @@ void MeshForwarder::HandleSentFrame(Mac::Frame &aFrame)
         {
             child->mFragmentOffset = 0;
             mSendMessage->ClearChildMask(mMle.GetChildIndex(*child));
+
+            if ((child->mMode & Mle::ModeTlv::kModeRxOnWhenIdle) == 0)
+            {
+                mMac.ClearSrcMatchAddressEntry(macDest);
+
```

Comment:
Currently there are no duplicate address check when `AddSrcMatchAddressEntry`, each message for SED (no matter for the same SED or different SED) would have an entry in the source match table.
If there are two messages queued for the same destination, there would be two address entry in the source match table.  This code should be fine for this case. 

FYI, below is the sample ping traffic in one topology (1 leader <->1 Router<->2SEDs,
, continuous ping mode from the leader/the router to one SED/RealmLocalAllThreadNodes with different payload 20/100)
[test_ping.pcapng.zip](https://github.com/openthread/openthread/files/469643/test_ping.pcapng.zip)

However problems would be raised if there are too many queued messages for SEDs.
Thanks.